templates/aes-injector.yaml

```diff
@@ -0,0 +1,140 @@
+{{- if and .Values.enableAES .Values.servicePreview.enabled .Values.injector.enabled }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "ambassador.fullname" . }}-injector
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: {{ include "ambassador.fullname" . }}-injector
+    app.kubernetes.io/part-of: {{ .Release.Name }}
+    helm.sh/chart: {{ include "ambassador.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    {{- if .Values.deploymentTool }}
+    app.kubernetes.io/managed-by: {{ .Values.deploymentTool }}
+    {{- else }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    {{- end }}
+    product: aes
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "ambassador.fullname" . }}-injector
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "ambassador.fullname" . }}-injector
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: webhook
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
```

Comment:
What `image:tag` is used here? Is the same as the AES Deployment? Could this confuse the installation detection in `edgectl install`?

Reply:
The same `image:tag` is used by the AES Deployment, `traffic-manager` and `aes-injector`. (Note it is also in use by the traffic-agent, referenced in the `TRAFFIC_AGENT_IMAGE` env var below).

I believe it could cause issues if Service Preview is installed by hand and AES is installed by `edgectl install`. I remember @ark3 planned for this scenario when building the existing install detection logic. @inercia could it also have an impact on `edgectl upgrade`?

Reply:
I'm concerned about all of these Deployemnts having the same 1) labels 2) `image:tag`. That could lead to some false possitives in `edgectl install/upgrade` when detecting previous installations, if one of those is present and another one is missing. But I think that would be a really uncommon case...

Reply:
> I'm concerned about all of these Deployemnts having the same 1) labels 2) `image:tag`

It's a very valid concern. However, I do believe the labels and image:tag values are the right ones. If it causes confusion, it certainly won't be fixed by generating new values in the chart itself, but rather through changes in `edgectl`.